src/pages/videopage/videopage.js

```diff
@@ -63,7 +89,48 @@ export default function Videopage() {
             })
         }
         dispatch({ type: "ADD_TO_PLAYLIST", payload: plLISTS });
-
+    }
+    const addToLikes = async () => {
+        if (user) {
+            const res = await axios.post("/api/user/likes", { video }, {
+                headers: {
+                    authorization: encodedToken
+                }
+            })
+            dispatch({ type: "ADD_TO_LIKES", payload: res.data.likes });
+        } else {
+            alert("Please login to add to likes.");
+            navigate("/login");
+        }
+    }
+    const removeFromLikes = async () => {
+        const res = await axios.delete(`/api/user/likes/${_id}`, {
+            headers: {
+                authorization: encodedToken
+            }
+        })
+        dispatch({ type: "REMOVE_FROM_LIKES", payload: res.data.likes });
+    }
+    const addToWatchLater = async () => {
+        if (user) {
+            const res = await axios.post("/api/user/watchlater", { video }, {
+                headers: {
+                    authorization: encodedToken
+                }
+            });
+            setUser({ ...user, watchlater: res.data.watchlater });
+        } else {
+            alert("Please login to add to watchlater.");
+            navigate("/login");
+        }
+    }
+    const removeFromWatchLater = async () => {
+        const res = await axios.delete(`/api/user/watchlater/${_id}`, {
+            headers: {
+                authorization: encodedToken
+            }
+        })
+        setUser({ ...user, watchlater: res.data.watchlater });
```

Comment:
suggestion : I would be good if we could take this functionality in a custom hook . Increases readability

Reply:
Noted the suggestions, will work on them. Thanks!